### PR TITLE
TINKERPOP-3090 TrimGlobalStep handles unicode characters as Ltrim and Rtrim

### PR DESCRIPTION
**Pull request by rdtr**
https://issues.apache.org/jira/browse/TINKERPOP-3090

The current behavior is inconsistent because it relies upon a different API to detect space characters. I think String#trim does not handle unicode characters.

```
gremlin> g.inject("\u3000abc\u3000").trim().next().length()
==>5
gremlin> g.inject("\u3000abc\u3000").lTrim().next().length()
==>4
gremlin> g.inject("\u3000abc\u3000").rTrim().next().length()
==>4
```

### CODE REVIEW

**Review by saikiranboga — APPROVED** [review]

File: CHANGELOG.asciidoc [hunk 1]
```diff
@@ -39,6 +39,7 @@ This release also includes changes from <<release-3-6-7, 3.6.7>>.
 * Deprecated `ltrim()` and `rTrim()` in favor of `l_trim()` and `r_trim` in Python.
 * Fixed bug in `onCreate` for `mergeV()` where use of the `Cardinality` functions was not properly handled.
 * Fixed multiple concurrent initially requests caused authentication to fail.
+* Fixed so that TrimGlobalStep has the same character control handling as Ltrim and Rtrim
```
Comment (spmallette) [inline CHANGELOG.asciidoc:42]: Looks like you need move your changelog entry up to 3.7.3

**Comment by spmallette** [comment]
VOTE +1, pending CHANGELOG fix

**Comment by xiazcy** [comment]
VOTE +1, also pending GLV feature test file updates (i.e. `gremlin.cs`, `gremlin.js`, `gremlin.py`, `gremlin.go`).

**Comment by vkagamlyk** [comment]
Good to also update `TrimLocalStep` to unsure consistent behavior.

File: gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/map/RTrim.feature [hunk 2]
```diff
@@ -19,11 +19,12 @@
 Feature: Step - rTrim()
 
   @GraphComputerVerificationInjectionNotSupported
+  # This verifies both ASCII control space and ideographic space character \u3000 are property trimmed.
   Scenario: g_injectX__feature___test__nullX_rTrim
     Given the empty graph
     And the traversal of
       """
-      g.inject("feature  ", "one test ", null, "", " ").rTrim()
+      g.inject("feature  ", "one test ", null, "", " ", "　abc", "abc　", "　abc　", "　　").rTrim()
```
Comment (vkagamlyk) [inline gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/map/RTrim.feature:27]: could you add test with `\u3000`?

Comment (rdtr) [reply gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/map/RTrim.feature:27]: I copied and tested `\u3000` on the test file. You may notice the 4 values I newly injected in the test contains a full-width space, not the ASCII control "regular" half-width space character. It did not work just to write as \u3000 on the Gherkin file.
Is there another proper approach to test unicode in Gherkin ?

Comment (vkagamlyk) [reply gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/map/RTrim.feature:27]: My bad, I didn't notice that there weren't just spaces there.

Probably simplest solution is to add test cases to [Java tests](https://github.com/apache/tinkerpop/blob/3.7-dev/gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/step/map/RTrimGlobalStepTest.java)

File: CHANGELOG.asciidoc [hunk 3]
```diff
@@ -30,6 +30,7 @@ This release also includes changes from <<release-3-6-8, 3.6.8>>.
 * Deprecated public constructor for `SeedStrategy` in favor of builder pattern to be consistent with other strategies.
 * Allow specifying a customized Spark app name
 * CoinStep has a getter method for its probability field
+* Fixed so that TrimGlobalStep has the same character control handling as Ltrim and Rtrim
```
Comment (vkagamlyk) [inline CHANGELOG.asciidoc:33]: nit: mention TrimLocalStep

**Comment by vkagamlyk** [comment]
VOTE+1

**Comment by rdtr** [comment]
> same character control

done! Now they all rely on the same common method

File: CHANGELOG.asciidoc (hunk 1 above)
Comment (rdtr) [reply CHANGELOG.asciidoc]: moved, thanks
